### ISSUE TITLE
folded lines with a different trailing character

### ISSUE THREAD

**Issue by teto** [neovim/neovim#13694]
sometimes, the last character on a folded line is highlighted incorrectly. I would like to fix it but haven't been able to reproduce it systematically. @lervag (or anyone else) if you know a way let me know

**Comment by lervag** [comment]
Thanks @teto, this is indeed an issue. I described it in https://github.com/neovim/neovim/pull/13393#issuecomment-734713039.

A minimal example that reproduces it for me:

**test.vim**

```vim
syntax enable
silent edit test.txt
normal! zM
```

**test.txt**

```
{{{ test 1

}}}
{{{ test 2

}}}
{{{ test 3

}}}

# vim: fdm=marker
```

When I do `nvim -u test.vim`, the incorrect highlight/glitch is visible along the right edges:

![2020-11-27_10:13_screenshot](https://user-images.githubusercontent.com/998898/100431992-4161f380-3099-11eb-942c-8687bcc9f9dd.png)

As I mentioned in the other thread: it seems this issue is related to the listschars. The glitch disappears if I do `set nolist`. See also `:help lcs-extends`.

If relevant: I use urxvt on Arch Linux in a 80 column terminal window.

**Comment by teto** [comment]
I can reproduce that's perfect thanks. What a valuable user :D do you use nvim or you are just checking because of vimtex user reports ?

**Comment by lervag** [comment]
I use neovim, but I'm a conservative user so I've still not had time to look into lua. I also plan to keep supporting Vim with vimtex for as long as possible, so I plan to stick with vimscript (which I know too well!).

**Comment by teto** [comment]
found it, I have a fix I will post later on. It's not related to list, Try `:hi Search guibg=red` :p

**Comment by teto** [comment]
https://github.com/neovim/neovim/pull/13704

**Comment by lervag** [comment]
Great! I'm very happy to see this fixed, as it has been a minor annoyance for quite some time :D

**Comment by teto** [comment]
this part of the code is especially tricky, I want to refactor it a bit but after 0.5. If you use the patch locally, let me know if you see something weird, I will do the same and merge later on

**Comment by lervag** [comment]
Will do! Btw: Although I can confirm the behaviour with the `Search` highlight group, I also find that the `NonText` highlight group is relevant. With the minimal example provided, only the `Search` group is relevant, but with my personal configuration, the `NonText` group is the relevant one. I have not had the time to investigate why as of yet, but I can do it if this means there are two separate issues here.

Notice the screenshots. In the first, I've changed `NonText` to have `ctermbg=3` and `Search` to have `ctermbg=4` (no `termguicolors` here). In the second, with `termguicolors`, I've changed `NonText` to `guibg=red` and `Search` to `guibg=blue`.

![2021-01-07_14:59_screenshot](https://user-images.githubusercontent.com/998898/103901292-3e770b80-50f9-11eb-99dd-7813f2a770f7.png)
![2021-01-07_15:00_screenshot](https://user-images.githubusercontent.com/998898/103901294-3f0fa200-50f9-11eb-919b-abc7c66a2e05.png)
